azkaban-exec-server/src/main/java/azkaban/execapp/metric/ProjectCacheHitRatio.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.execapp.metric;
+
+import com.codahale.metrics.RatioGauge;
+import com.codahale.metrics.SlidingWindowReservoir;
+
+/**
+ * Project cache hit ratio of last 100 cache accesses.
+ */
+public class ProjectCacheHitRatio extends RatioGauge {
+
+  private final SlidingWindowReservoir hits;
+  private final SlidingWindowReservoir calls;
+  public static final int WINDOW_SIZE = 100;
```

Comment:
100 seems small, for cases where there are a few thousand executions per day. It would be good if this is a configurable parameter (at some point, it would be good to be able to deploy configurations without a complete Azkaban deployment), or if it is time-based (last hour, etc.).

Reply:
explained in class comment.